Algorithm.Python/Alphas/MultipleShareClassMeanReversionAlpha.py

```diff
@@ -0,0 +1,241 @@
+    mean-reversion. Using a Simple Moving Average indicator, we can
+    compare the value of this portfolio against its SMA and generate insights
+    to buy the under-valued symbol and sell the over-valued symbol.
+'''
+
+
+from clr import AddReference
+AddReference("System")
+AddReference("QuantConnect.Algorithm")
+AddReference("QuantConnect.Common")
+AddReference("QuantConnect.Indicators")
+
+from System import *
+from QuantConnect import *
+from QuantConnect.Algorithm import *
+from QuantConnect.Data.Market import TradeBar
+from QuantConnect.Algorithm.Framework import *
+from QuantConnect.Algorithm.Framework.Risk import *
+from QuantConnect.Orders.Fees import ConstantFeeModel
+from QuantConnect.Algorithm.Framework.Alphas import *
+from QuantConnect.Algorithm.Framework.Execution import *
+from QuantConnect.Algorithm.Framework.Portfolio import *
+from QuantConnect.Algorithm.Framework.Selection import *
+from QuantConnect.Indicators import RollingWindow, SimpleMovingAverage
+
+from datetime import datetime, timedelta
+import pandas as pd
+import numpy as np
+
+class ShareClassMeanReversionAlphaModel(QCAlgorithmFramework):
+
+    def Initialize(self):
+
+        # Set requested data resolution
+        self.UniverseSettings.Resolution = Resolution.Minute
+
+        self.SetStartDate(2019, 1, 1)   #Set Start Date
+        self.SetCash(100000)           #Set Strategy Cash
+        self.SetWarmUp(20)
+        
+        ## Set the universe resolution and select a basket of dual-class shares
+        self.UniverseSettings.Resolution = Resolution.Minute
+        tickers = ['VIA','VIAB','GOOG','GOOGL','COKE','KO','LEXEA','EXPE','TRIP','LTRPA']
+        ticker_pairs = [['VIA','VIAB'],['GOOG','GOOGL'],['COKE','KO'],['LEXEA','EXPE'],['TRIP','LTRPA']]
+        
+        ## Create Symbols for manual universe selection
+        symbols = [ Symbol.Create(ticker, SecurityType.Equity, Market.USA) for ticker in tickers]
+
+        self.SetUniverseSelection( ManualUniverseSelectionModel(symbols) )
+
+        ## Set $0 fees
+        self.SetSecurityInitializer(lambda security: security.SetFeeModel(ConstantFeeModel(0)))
+        
+        self.SetAlpha(ShareClassMeanReversionAlphaModel(ticker_pairs = ticker_pairs, number_pairs = len(ticker_pairs), ticker_list = tickers))
+        
+        self.SetPortfolioConstruction(EqualWeightingPortfolioConstructionModel())
+        
+        self.SetExecution(ImmediateExecutionModel())
+        
+        self.SetRiskManagement(NullRiskManagementModel())
+        
+
+
+class ShareClassMeanReversionAlphaModel(AlphaModel):
+
+    def __init__(self, *args, **kwargs):
+        ''' Initialize variables and dictionaries to help with SMA and Symbol Date
+        object creation/maintenance'''
+        self.symbolDataBySymbol = {}
+        self.sma_period = 15
+        self.period = timedelta(minutes=5)
+        self.number_pairs = kwargs['number_pairs'] if 'number_pairs' in kwargs else None
```

Comment:
`self.ticker_list`, `self.ticker_pairs` and `self.number_pairs` cannot default to `None`. For the last attribute, we can use `len(self.ticker_pairs)` instead:
```python
if 'ticker_pairs' not in kwargs:
    raise Exception('ShareClassMeanReversionAlphaModel: Missing argument: "ticker_pairs"')
self.ticker_pairs = kwargs['ticker_pairs']
self.number_pairs = kwargs['number_pairs'] if 'number_pairs' in kwargs else len(self.ticker_pairs)
```